File: Array/MaxSunContiguousSubarray/MaxSumContiguousSubarray.py
class Solution:
    # @param A : tuple of integers
    # @return an integer
    # Time Complexity  : O(n)
    # Space Complexity : O(1)
    def maxSubArray(self, A):
        res = A[0]
        temp = A[0]
        for val in A[1:]:
            temp = max(val,temp+val)
            res = max(temp,res)
        return res

File: Array/MaxSunContiguousSubarray/MaxSumContiguousSubarray.cpp
/* 
Time Complexity  : O(n)
Space Complexity : O(1)
*/
int Solution::maxSubArray(const vector<int> &A) {
    int res = A[0],temp=A[0];
    for(int i=1;i<A.size();i++){
        temp = max(A[i],temp+A[i]);
        res = max(res,temp);
    }
    return res;
}
